### ISSUE TITLE
about similarity as distance in triplet loss function

### ISSUE THREAD

**Issue by KennyTC** [KevinMusgrave/pytorch-metric-learning#454]
In the triplet loss function, 
- if we use Euclidean distance as the distance, then the triplet loss function will be [d_ap - d_an + margin]+, 
- if we use cosine similarity as the distance, then the triplet loss function will be [s_an - s_ap + margin]+, according to [this](https://kevinmusgrave.github.io/pytorch-metric-learning/distances/#cosinesimilarity).

In the case of Euclidean distance, the distance is always positive, so I can understand the above triplet loss function.
However, in the case of cosine similarity, since cosine similarity is [-1, 1], ie. it can be negative and positive. so I wonder if [s_an - s_ap + margin]+ is true for both case (cosine similarity is negative and positive)?

**Comment by KennyTC** [comment]
thanks, it is clear now.

**Comment by Egorignatenko** [comment]
ср, 23 мар. 2022 г., 16:41 KennyT ***@***.***>:
